tests/acceptance/test_build.py

```diff
@@ -510,11 +521,11 @@ def test_preferred_versions(self, prepared_test_build, recipe, version):
 
             run_verbose("%s && bitbake %s" % (init_env_cmd, recipe))
 
-        # QA-784: setting/unsetting PREFERRED_VERSION and rebuilding the recipes breaks some times
-        # the state cache so that consecutive tests will fail in a very inexplicable way.
-        # Cleaning the cache here for each version built seems to workaround the issue...
-        run_verbose("%s && bitbake --cmd cleanall %s" % (init_env_cmd, recipe))
+        # Clean leftovers to have a fresh start for the next recipe
+        run_verbose("%s && bitbake --cmd clean %s" % (init_env_cmd, recipe))
+
 
+class TestBuildMoreTODOMove:
```

Comment:
Forgotten?

Reply:
I'll rework after your feedback but I wanted to make absolutely sure that the tests were being ordered the same way (hence I split this class into two) to validate the approach. Ref my general comment above.